DNS'25 feedback

Multipacker opened the pull request:
Vi har uppdaterat boken med nytt material under sommaren och skulle uppskatta feedback! Detta är tänkt som ett ställe att samla all feedback på för att underlätta för oss.

Multipacker committed:
Made custom command work when you switch OS

Multipacker committed:
Made custom commands escapable

Multipacker committed:
Expand documentation for contributing

Multipacker committed:
Fix mailto links, remove dates for lectures and move CLI section before git

Multipacker committed:
We accidentally included the command still, whoops

SwedishSubmarine committed:
Updated authors

SwedishSubmarine committed:
Changes to 2. Editor

Added some text to the pros for VSCode, additionally rewrote the final
paragraph a bit to reflect the future sub-chapters.

Multipacker committed:
Fix footnotes

Multipacker committed:
No more analytics

SwedishSubmarine committed:
Basics chapter for editor.

Some of this feels a bit clumsy... 🙃

SwedishSubmarine committed:
mega minor change 🐈

SwedishSubmarine committed:
A sentence was moved to the basics subchapter

Multipacker committed:
Allow for slightly more flexible syntax in custom commands

Multipacker committed:
More contributing documentation

SwedishSubmarine committed:
Changed some things in basic

Added linux folder creation (only for gnome because I am too lazy to
test every single file manager) and fixed some formatting

SwedishSubmarine committed:
Moved shortcuts to be after navigation

SwedishSubmarine committed:
Rewrote the shortcuts section a bit

Also added the cheatsheet in as a png so it's available on the page and
not just as a download. Also i hate gimp i hate gimp i hate gimp i hate
gimp

SwedishSubmarine committed:
Added some information around end + start of line

SwedishSubmarine committed:
Added note about choosing OS in getting started

Multipacker committed:
Fix warning about not selecting an OS or programme

SwedishSubmarine committed:
in my defense it was a bug

SwedishSubmarine committed:
Made the ctrl or cmd box specific to macos

SwedishSubmarine committed:
secret - just guess 🐱

anyways i changed three titles to be not big titles but less big titles
ok

SwedishSubmarine committed:
uhh i think i added the note about activity bar

it's been a few hours ok

SwedishSubmarine committed:
Added the extensions sub-chapter

It's a bit long maybe :/
I'm not sure how to cut it down so open for ideas

SwedishSubmarine committed:
Adding the extensions sub-chapter into the sidebar

SwedishSubmarine committed:
Minor text changes, testing VSCode git

SwedishSubmarine committed:
Updating mdbook image version

surely this will be fine :unaware: 🐈 🙃

Multipacker committed:
First step in rewriting the CLI chapter

Multipacker committed:
Added a quick refernence for commands with short explanation of each

Multipacker committed:
Fixed minor inconsistencies in the editor chapter

Multipacker committed:
Restructure the CLI intro

Multipacker committed:
Added notes about hidden files and arguments with spaces in them

SwedishSubmarine committed:
Minor changes to index of CLI

Multipacker committed:
Split the CLI chapter to an installation part and a basics part

SwedishSubmarine committed:
Added a list for why to use the cli

VSCode merge conflict testing ^^

Multipacker committed:
Try to better match the original styles to make it easier to update in the future, also fix bug with table of contents

SwedishSubmarine committed:
Git - index

SwedishSubmarine committed:
here be package managers

Multipacker committed:
Make theme more similar to default

Multipacker committed:
Merge in autocompletion into terminal basics

SwedishSubmarine committed:
Moved a bunch of stuff around, SSH and remote

SwedishSubmarine committed:
??? i changed some stuff ok

Multipacker committed:
Fixed weird spacing around headers

Multipacker committed:
Expand on package managers for Linux

Multipacker committed:
Switched to using a table for listing commands

Multipacker committed:
Fix the order of sub-chapters in CLI

Multipacker committed:
Updated the chapter on command-line flags

SwedishSubmarine committed:
Added some more information about homebrew

SwedishSubmarine committed:
Added headers to make installation easier to read

SwedishSubmarine committed:
Added key setup on remotes

Multipacker committed:
Minor fixups to chapter on environment variables

Maybe this should be expanded upon more.

Multipacker committed:
Added notes about how PowerShell is different from Unix shells

SwedishSubmarine committed:
Changed some minor stuff

SwedishSubmarine committed:
Git basics, not super happy with this yet

needs feedback

SwedishSubmarine committed:
Added how to commit to git in terminal 🐱

yes i know it's called teminal.png whatcha gonna do about it

SwedishSubmarine committed:
Realized this chapter is long

SwedishSubmarine committed:
Added how to use VSCode and terminal with a remote

I also commented out merge conflict information specific to GitKraken.
This should probably move to a separate sub-chapter because it is taking
up a lot of space but then we'll have 8 sub-chapters here which is quite
a lot.

SwedishSubmarine committed:
Rewrote parts of remote for clarity and structure

SwedishSubmarine committed:
minor changes

SwedishSubmarine committed:
Added note about quitting editors

SwedishSubmarine committed:
VSCode as default editor

SwedishSubmarine committed:
Added merge conflicts back in

Added how to handle merge conflicts in a terminal

Multipacker committed:
Feedback and minor fixes for Git chapters 4.1 through 4.4

Multipacker committed:
Added some extra notes about how commands behave if they encounter errors

Multipacker committed:
Fixed a bug where one extra character would disappear when using `!icon`

SwedishSubmarine committed:
Changed an image to be less wack

SwedishSubmarine committed:
Rewrote some text to be more clear

SwedishSubmarine committed:
Not done but got rid of gitkraken

SwedishSubmarine committed:
Minor changes and note about ed25519

SwedishSubmarine committed:
Warning about vi / nano

Multipacker committed:
Mention that you can also open a terminal inside of VSCode

Multipacker committed:
Added more notes on conventions for command help pages and a small section on using `sudo`

SwedishSubmarine committed:
Information on package managers on windows

SwedishSubmarine committed:
Forgot to push this up 🙅

SwedishSubmarine committed:
liveshare, some to-dos i need a second person to help test

SwedishSubmarine committed:
fixes for studies

SwedishSubmarine committed:
masters programme information

SwedishSubmarine committed:
AI chapter added, quite lengthy

SwedishSubmarine committed:
forgot to update general tips link

SwedishSubmarine committed:
clarifications

Multipacker committed:
Minor formatting and spelling fixes to AI chapter

Multipacker committed:
Fixed spelling in AI draft

Multipacker committed:
First attempt at describing files and the file system

Multipacker committed:
Short description of canvas

SwedishSubmarine committed:
moving files to the correct place

Multipacker committed:
Started on Chalmers Apps

Multipacker committed:
Center images inside the main element

Why is this not the default...

Multipacker committed:
Rewrote the SSH chapter a bit

Multipacker committed:
`!kbd` and `!icon` commands no longer display `undefined`

SwedishSubmarine committed:
new information about masters

SwedishSubmarine committed:
?

SwedishSubmarine committed:
git hates me

Multipacker committed:
Expanded a bit on SSH keys and rewrote other part of SSH

Multipacker committed:
Added AppsAnywhere and soooooo many pictures

SwedishSubmarine committed:
Changed a sentence in AI

SwedishSubmarine committed:
Minor formatting change

SwedishSubmarine committed:
Formatting and language

SwedishSubmarine committed:
RIP Virtualization

SwedishSubmarine committed:
Rewrote portions for 25/26

SwedishSubmarine committed:
Added references to new chapters and chalmers apps

SwedishSubmarine committed:
Formatting

SwedishSubmarine committed:
no one saw that

SwedishSubmarine committed:
Editor formatting

SwedishSubmarine committed:
CLI formatting and minor additions

SwedishSubmarine committed:
moving to my desktop

SwedishSubmarine committed:
Git formatting and small changes

SwedishSubmarine committed:
Git formatting and minor changes

SwedishSubmarine committed:
Got rid of a weird sentence

SwedishSubmarine committed:
Formatting AI

SwedishSubmarine committed:
Restructured so index and sub-chapters are first under studies

Multipacker committed:
Allow you to only include footnotes for specific platforms

This solution is not perfect. If all footnotes are hidden, you still get
the separation line between the footnoes and the main text.

Multipacker committed:
Update instructions for LiveShare

Multipacker committed:
You can now have images and non-text that is specific to a platform

Also cleaned up the code a bit

Multipacker committed:
More cleanup to custom pre-processor

Multipacker committed:
Minor fixes to AppsAnywhere

Multipacker committed:
Consistency treatment for Canvas section

Multipacker committed:
Minor changes to advanced editor

Multipacker committed:
Fixed minor grammar and consistency things in introduction

Multipacker committed:
Fixed minor grammar and consistency things in editor and liveshare chapters

Multipacker committed:
No more trailing whitespace:)

Multipacker committed:
Update instructions for how to run the book locally and include them in the readme

Multipacker committed:
Minor fixes to file-system chapter

Multipacker committed:
When explaining package managers, hide the headings if you only have one operating system selected

Multipacker committed:
Minor fixes to CLI chapter and formatting of text

Hugsim commented on src/studies/exams.md:
```diff
@@ -20,14 +20,15 @@ As a student, you have certain rights and obligations which you should be aware
 ## After the exam
 
-The examiner **must** post solutions to the exam by the next working day and the results should be posted on Ladok within 15 working days.
+The examiner **must** post solutions to the exam by the next working day and the results should be posted on Ladok within 15 working days. If the results are not posted by this time, contact the examiner, and if you can not get an answer from them as to why, contact your student board!
```
Ni har använt begreppet "Studienämnd" inom citattecken på engelska innan, var konsekventa. Om ni vill använda det engelska begreppet brukar studienämnder översättas till "Student Educational Committee".

Hugsim commented on src/courses/AI.md:
Skulle det kanske vara rimligt att länka till reglerna för att använda AI som Chalmers har? En snabb googling ger mig https://www.chalmers.se/utbildning/dina-studier/kandidat-och-examensarbete/regler-for-anvandning-av-ai-verktyg/ och https://www.chalmers.se/en/education/your-studies/masters-and-bachelors-thesis/regulations-for-the-use-of-ai-tools/ för kandidat-/examensarbeten. 

Åtminstone borde det beskrivas tydligt att det är upp till varje examinator att bestämma i varje kurs vad som gäller.

SwedishSubmarine replied:
Ja absolut det ska vi ha med! Ska försöka komma ihåg att uppdatera det här när Chalmers uppdaterat sin AI-policy. Inte helt säker på när den ska vara färdig men vi har diskuterat den lite med Magnus och den är väldigt WIP just nu så har med den länken för tillfället.

Hugsim commented on src/studies/courses.md:
```diff
@@ -10,7 +10,7 @@ These are the courses you have no say over and are deemed essential for your deg
 Sometimes you have elective slots in your program plan, which means that you can select any course marked as _elective_ within your program plan.[^elective_caveat] For master students it may be interesting to know that you can read one or two courses (around 15hp) outside of your program plan and still get your masters degree. This can vary between programs so ask your **student counsellor** befo ...
 
-You can find your on [Studieportalen](https://www.student.chalmers.se/sp/program_list) by searching for your programme.
+You can find your program plan on [Studieportalen](https://www.student.chalmers.se/sp/program_list) by searching for your programme.
```
Chalmers översätter programplan till "programme overview" som ni borde ändra till i det här kapitlet.

Multipacker replied:
Bra fångat!

Hugsim commented on src/studies/index.md:
```diff
@@ -1,10 +1,15 @@
+This chapter serves the purpose to give you information which might be important
+during your time at Chalmers. This includes your rights as a student and what
+you can do to affect your studies. We will also include general information
+regarding studying at a swedish university.
```
```suggestion
regarding studying at a Swedish university.
```

Hugsim commented on src/studies/master_old.md:
Samma kommentar som för det nya kapitlet om mastern gäller här med, mastersprogram -> master's programme

Hugsim commented on src/studies/master.md:
Chalmers översätter masterprogram till "master's programme", föreslår att ni är konsekventa i detta kapitlet.

Hugsim reviewed (commented):
Snyggt jobbat! Bara några små kommentarer på lite stavningar. Har också en kommentar på AI-stycket, feel free att ignorera den om ni inte håller med. :)

Multipacker committed:
Fixed most of the feedback from @Hugsim

The1Penguin commented on src/beginner/files.md:
Jag är lite ledsen att denna behövs, men så är livet. Tycker exemplen fungerar bra, och inte går in på djupet så det blir negativt. Det skulle vara nice om flowet går bra in till nästa, men det kan vara svårt

Multipacker replied:
Flowet från ett exempel in i nästa då?

The1Penguin replied:
Nej, menade till nästa kapitel, dvs editor

The1Penguin commented on src/beginner/editor/index.md:
```diff
@@ -6,14 +6,13 @@ The tool you will be looking at the most when working on your computer will be e
-
-The only editor which truly fits all of these is Visual Studio Code.
+- Extensive plugin support for different languages and for use in certain courses at Chalmers.
+- Very configurable to your needs with a simple to understand GUI. (**G**raphical **U**ser **I**nterface)
```
```suggestion
- Very configurable to your needs with a simple to understand **GUI**. (**G**raphical **U**ser **I**nterface)
```

The1Penguin commented on src/beginner/editor/index.md:
```diff
@@ -22,12 +21,10 @@ To get started, begin by downloading VSCode from the [webpage](https://code.visu
 > **Did you know that you can open links in a new tab by either**
 >
 > - Holding !kbd[!ctrl] while clicking on the link
-> - Middle click on the link
+> - Middle clicking on the link
```
Kanske ändra till mer passiv röst?

SwedishSubmarine replied:
Förstår inte riktigt hur du menar

The1Penguin replied:
nvm, tänkte en sak baserat utan att hålla koll på raden innan

The1Penguin commented on src/beginner/editor/basics.md:
```diff
@@ -0,0 +1,79 @@
+quick overview of the very basic functionality of your new editor. In the top
+left corner you will see an arrow titled "Welcome", this will take you back to
+the _Welcome_ page which greets you every time you start VSCode from here on
+out.
```
Kanske ha bild för att visa, ifall saker ändrar sig? Vad menas med följande?
```
you will see an arrow titled "Welcome"
```

SwedishSubmarine replied:
Lägger till en bild och skriver om lite för tydlighet

The1Penguin commented on src/beginner/editor/basics.md:
```diff
@@ -0,0 +1,79 @@
+![The welcome page](/Assets/editor/welcome.png)
+
+The one we're going to start with is `Open...` or `Open Folder...`.
+Alternatively using your keyboard press !kbd[!ctrl+O](note that it is the
+letter O, not zero). This will bring up your operating systems graphical file
```
```suggestion
Alternatively using your keyboard press !kbd[!ctrl+o](note that it is the
letter, not zero). This will bring up your operating systems graphical file
```

SwedishSubmarine replied:
FINE :cat:

SwedishSubmarine replied:
have now fixed for every single one. in the book. this sucked.

The1Penguin commented on src/beginner/editor/basics.md:
```diff
@@ -0,0 +1,79 @@
+
+> This warning goes out especially to Windows users, but does apply to
+> **everyone**. Do not use spaces in your file or folder names. At best it will
+> be an annoyance and at worst it will break things!
```
```suggestion
> This warning goes out to Windows users especially, but does apply to
> **everyone**. Do not use spaces in your file or folder names. At best it will
> be an annoyance and at worst it will break things!
```

The1Penguin replied:
Också, kanske flytta till files?

SwedishSubmarine replied:
Det här är första gången vi nämner att faktiskt skapa filer. Vi flyttar upp den lite men behåller den i editor. Det blir väldigt mycket varningar och text blocks i `files` annars.

The1Penguin commented on src/beginner/editor/basics.md:
```diff
@@ -0,0 +1,79 @@
+
+{{ .windows }} On Windows you will most likely be placed in your user folder.
+To create a new folder, right click and hover over `New`, here select `Folder`.
+Name it something like "data101" and then press `Select Folder`.
```
"Our" istället för "My"

SwedishSubmarine replied:
Thankie :pleading_face:

The1Penguin commented on src/beginner/editor/basics.md:
```diff
@@ -0,0 +1,79 @@
+you can press !kbd[!ctrl+N] which immediately creates an untitled text file.
+You will then get to name it once you save it.
+
+> Left of the Explorer you can see the so called _Activity Bar_, by right
+> clicking this you can change its contents or position on the screen.
```
Bild för tydlighet?

SwedishSubmarine replied:
Hade en från början men väldigt vertikala bilder ser hemska ut så tog bort den igen. Eftersom det här mest är en kommentar och inte kritisk kunskap så tycker jag det är fine utan bild.

The1Penguin replied:
Isåfall skulle jag göra en bild med en pil. Jag förstod inte vad som var activity bar (dock inte en vscode enthusiast).
Jag skulle dock påstå att det är vettigt att nämna, med tanke på att ni refererar tillbaka till det senare i extensions delen.

SwedishSubmarine replied:
La till en!

The1Penguin commented on src/beginner/editor/navigation.md:
```diff
@@ -1,56 +1,124 @@
+about it but the name may be new to you. To move the caret, use the arrow keys.
+If your keyboard has them, you can also move to the start or end of the line by
+pressing !kbd[Home] or !kbd[End] respectively. To move the caret word by word,
+hold !kbd[Ctrl] (or !kbd[Alt] on MacOS) while pressing !kbd[Left] or
```
```suggestion
hold !kbd[Ctrl] (or !kbd[Option] on MacOS) while pressing !kbd[Left] or
```
Är detta rätt?

SwedishSubmarine replied:
Spanner fix :+1:

The1Penguin commented on src/beginner/editor/navigation.md:
```diff
@@ -1,56 +1,124 @@
+There are four main modes of operation. The first and simplest one is to search
+for files in your project. Try creating a new file in your project, open the
+command palette and then type the beginning of the name and press !kbd[Enter].
+This is a fast method of moving between files in your project.
```
Lägg till förslaget att navigera här också

SwedishSubmarine replied:
Kommer du ihåg vad det här refererar till?

The1Penguin replied:
nope

Multipacker replied:
Stänger denna eftersom vi inte kommer ihåg vad den refererar till.

The1Penguin commented on src/beginner/editor/navigation.md:
```diff
@@ -1,56 +1,124 @@
+do by using your mouse can be done here. Try opening a file using the command
+palette! This mode can be used for almost anything you can think of. When you're
+in this _Command_ mode try searching for something you want to do. Be it "Create
+new file" or "Add line Comment", the command palette can probably do it!
```
```suggestion
do by using your mouse can be done here. When you're in this _Command_ mode try searching for something you want to do. Be it "Create
new file" or "Add line Comment", the command palette can probably do it!
```

The1Penguin commented on src/beginner/editor/navigation.md:
```diff
@@ -1,56 +1,124 @@
-When writing text you seldom worry about adding line breaks, but in code lines have _semantic_ meaning. They change which computer instructions the code represents. Naturally because of this we have ways of working with lines.
+When writing text you seldom worry about adding line breaks, but in code lines
+have _semantic_ meaning. They change which computer instructions the code
+represents. Naturally because of this we have ways of working with lines.
```
Varför nämna semantik? Är det vedertaget för en ny student?

The1Penguin commented on book.toml:
```diff
@@ -1,5 +1,5 @@
 [book]
-authors = ["David Hedgren", "Love Lyckaro"]
+authors = ["David Hedgren", "Love Lyckaro", "Emily Tiberg", "Simon Renhult"]
```
```suggestion
authors = ["David Hedgren", "Love Lyckaro", "Nor Führ", "Erik Örtenberg", "Emily Tiberg", "Simon Renhult"]
```

The1Penguin commented on src/beginner/cli/index.md:
```diff
@@ -1,80 +1,74 @@
+- If you ever need to use a server, or a remote connection you will be
+  restricted to only using a terminal in most cases.
+- It's fun! It might seem scary at first but when you get used to it and learn
+  it, it can actually make using your computer a whole lot of fun.
```
Kanske nämna att det lättare att få instruktioner? Det overlappar med vän delen men känns bättre sagt

The1Penguin commented on src/beginner/cli/index.md:
```diff
@@ -1,80 +1,74 @@
+
+Most systems you will interact with use what is called a Unix terminal. Windows
+has its own terminals `PowerShell` and `cmd.exe`. These are quite different
+from other terminals, and so we will not discuss them here.
```
```suggestion
from other terminals, and as such will not be discussed here.
```

The1Penguin commented on src/beginner/cli/index.md:
```diff
@@ -1,80 +1,74 @@
+
+> A lot of people naturally get stressed and think that it is easy to write
+> some command that will cause harm to the computer. Please be assured by that
+> it is in fact incredibly difficult to destroy anything using the terminal.
```
```suggestion
> A lot of people naturally get stressed and think that it is easy to write
> some command that will cause harm to the computer. Please be assured that
> it is in fact incredibly difficult to destroy anything using the terminal.
```

The1Penguin commented on src/beginner/cli/index.md:
```diff
@@ -1,80 +1,74 @@
+is really old and is not used much these days. `PowerShell` is Microsofts newer
+replacement, and although it can be used on other platforms than Windows, the
+most commonly used terminals are still Unix based, and most command line
+utilities are written for Unix terminals.
```
Repeat? Flytta upp det som är nytt?

The1Penguin commented on src/beginner/cli/basics.md:
```diff
@@ -0,0 +1,121 @@
+
+> You can autocomplete commands and arguments by pressing !kbd[Tab]. If there
+> are multiple alternatives, you will get a list of them. You might need to
+> double-tap !kbd[Tab] to see all available alternatives.
```
Inte i Bash, vilket är default för många

SwedishSubmarine replied:
Är du säker? Jag tror inte att det är lika extensive som ZSH men tror det finns något by default.

The1Penguin replied:
Double tap fungerar inte på samma sätt mellan shells, I guess att detta stämmer, men det matchar inte hur många (från min uppfattning) förväntar sig att det ska fungera (likt en editor där tab byter mellan alternativ). Det är också att ifall bash får många alternativ så kan man få scenarion där det beter sig som `less` vilket är weird

SwedishSubmarine replied:
Ah jag förstår. Kanske är värt att skriva ett litet kapitel om shells under further reading som kan ha med lite mer information om auto completion för det är väldigt nice men det blir ganska mycket att skriva här särskilt eftersom vi inte riktigt har en bra introduktion till vad olika shells är och vad skillnaderna är.

The1Penguin replied:
Förstår det, jag tror att detta är mer av en situation där jag inte kan fokusera på helheten pga en detalj

The1Penguin commented on src/beginner/cli/basics.md:
```diff
@@ -0,0 +1,121 @@
+It is common to alternate between `cd` and `ls` when navigating in the
+terminal, but this can quickly clutter up the screen. You can clear it by
+either running the command `cls` (**cl**ear **s**creen), or by pressing
+!kbd[Ctrl+L].
```
`cls`???

The1Penguin commented on src/beginner/cli/environment_variables.md:
Ser ingen poäng här just nu

Multipacker replied:
Vi bestämde oss för att kasta bort kapitlet helt eftersom vi inte använder dem så mycket just nu

The1Penguin commented on src/beginner/cli/commands.md:
````diff
@@ -0,0 +1,44 @@
+sudo cp /etc/fstab /etc/fstab.copy
+```
+
+After pressing !kbd[Enter], depending on how your system is configured, you will be prompted for your users password. While typing it in, you will notice no characters are appearing on screen. Don't worry, your typing is being registered, this is just another security measure to make sure no one is peaking over your shoulder and sees the password. Once you have typed in your password, press !kbd[ ...
+
+You can configure which users can use `sudo` and which commands they are allowed to run with it in order to limit the access they get to your system. Commonly it is setup so you can run any command with `sudo`, provided you can authenticate yourself with your password.
````
Ni borde nämna att man ska vara mer försiktig med sudo

SwedishSubmarine replied:
Lägger till en kommentar.

The1Penguin commented on src/beginner/cli/powershell.md:
Where is?

Multipacker replied:
Är nu omskrivet lite och tillagt i slutet av CLI kapitlet. Osäker på om man vill gå igenom det mer genomgående eftersom folk är nya till det eller om det funkar så här

The1Penguin commented on src/beginner/git/setup.md:
```diff
@@ -1,118 +1,94 @@
+Now you'll be asked to write in a passphrase. Again, do ***not*** write
+anything here (unless you really know what you're doing). Certain programs like
+VSCode do not play nice with SSH-keys with a passphrase and can straight up
+crash with no warnings. Just press Enter, and when prompted to write the same
```
```suggestion
crash with no warnings. Just press !kbd[Enter], and when prompted to write the same
```

The1Penguin commented on src/beginner/git/basics.md:
```diff
@@ -33,22 +38,67 @@ a meaningful change. A good rule of thumb is to commit your changes each time
+> You can also create a so called README in your git repo if you'd like. A
+> README.md file is a text file which can be displayed on a remote right under
+> your files. This makes it valuable for documentation or information about your
+> project.
```
Ni har inte nämnt tydligt nämnt vad en remote är

SwedishSubmarine replied:
Flyttade till remote sub-kapitlet.

The1Penguin commented on src/beginner/git/basics.md:
```diff
@@ -33,22 +38,67 @@ a meaningful change. A good rule of thumb is to commit your changes each time
+mouse after you've saved the file.
+
+> [Warning]
+> If you don't change this your default editor will likely be nano or vi. If this is `vi` or `vim`, press `i`, type in your commit message and then press `Escape` and write `:wq` on your keyboard, this saves and quits your file and commits it. If this is `nano` write in your commit message and use `!kbd[ctrl+x]` to quit and then press `Y` to save the buffer.
```
```suggestion
> If you don't change this your default editor will likely be `nano` or `vi`. If this is `vi` or `vim`, press `i`, type in your commit message and then press `Escape` and write `:wq` on your keyboard, this saves and quits your file and commits it. If this is `nano` write in your commit message and use `!kbd[ctrl+x]` to quit and then press `Y` to save the buffer.
```

The1Penguin commented on src/courses/AI_draft.md:
Bort vid merge?

The1Penguin commented on src/courses/AI.md:
```diff
@@ -0,0 +1,121 @@
+> yourself.
+
+You might hear companies talking about how they want developers who know how to
+use AI. This is true—but what they mean is using AI as a tool, not a crutch.
```
Byt språkval. Använde inte crutch på det sättet

The1Penguin commented on src/courses/AI.md:
Ni behöver vara tydligare med vad som har påverkats av AI här, annars känns det inte riktigt okej

SwedishSubmarine replied:
Yea vi håller med. Har gjort om hur vi hanterar det.

The1Penguin commented on src/courses/AI.md:
Visa skillnaderna här, inte i en annan fil

The1Penguin reviewed (changes requested):
Ser bra ut för det mesta, kolla på mina kommentarer

SwedishSubmarine committed:
consistency with small letters in ctrl for keyboard shortcuts

SwedishSubmarine committed:
pingu basics comments 🐈

SwedishSubmarine committed:
**GUI**

Multipacker committed:
Added `!alt` to `kbd` command with custom display for MacOS

Multipacker committed:
Fixed the color of warning blocks

SwedishSubmarine committed:
small letters in keybinds YAYYY

Multipacker committed:
Switched to using the new `!alt` for keyboard shortcuts

SwedishSubmarine committed:
Activity bar picture

SwedishSubmarine committed:
Pingu navigation fixes

SwedishSubmarine committed:
Update authors properly this time

SwedishSubmarine committed:
Fixes for CLI

SwedishSubmarine committed:
Fixes for git

SwedishSubmarine committed:
Massive changes for AI

Got rid of the gimmick*. The main chapter should be clear and written by
us. Instead added a dropdown view so you can see a reformatted text
written by AI. Also added references to chalmers page on AI for the
bachelors thesis. This will most likely be rewritten sometime in the
next two years so this chapter will be updated when chalmers actual AI
policy comes out.

SwedishSubmarine committed:
raaa

Multipacker committed:
Fixed minor things in Git chapter

Multipacker committed:
Made sure the example merge conflict has the correct number of markup characters

Multipacker committed:
Explain parent directories more precisly

Multipacker committed:
Formatting in git chapter

Multipacker committed:
`!ctrl` be use a capital first letter when not generating for MacOS

Multipacker committed:
Started writing on eduroam

Multipacker committed:
Added theming for HTML details elements

Multipacker committed:
Expanded the section on how to connect to Eduroam from Linux with two tested approaches

Multipacker committed:
Rewrote PowerShell chapter and integrated it into the book

Multipacker committed:
Removed chapter on environment variables

Multipacker committed:
We no longer cover all four modes of the command palette, so reformulate to not confuse readers

Multipacker committed:
Removed some old unreferenced files and collected future ideas into TODO.md

Multipacker committed:
Clean up assets

SwedishSubmarine committed:
Updated TODO slightly

SwedishSubmarine committed:
Eduroam for mac

Multipacker committed:
Reworded some of git remote and introduced the shorthand repo in the begining of the chapter

Multipacker committed:
Updated git workflow to not mention GitKraken anymore

The1Penguin commented on src/beginner/editor/basics.md:
```diff
@@ -0,0 +1,88 @@
+# The Start Screen
+
+The first time you start up VSCode you will be presented with a short
+introduction, which can help you set up Copilot, choose a default theme, and
+that can give you resources like video tutorials for getting started.
+
+![Setup VSCode](/assets/editor/intro.png)
+
+However to not get stuck with customizing and watching videos we will give a
+quick overview of the very basic functionality of your new editor. In the top
+left corner you will see an small arrow with a "Welcome" next to it, this will
+take you back to the _Welcome_ page which greets you every time you start VSCode
+from here on out.
```
Bra med nya bilden

The1Penguin commented on src/beginner/editor/basics.md:
```diff
@@ -0,0 +1,88 @@
+you can press !kbd[!ctrl+n] which immediately creates an untitled text file.
+You will then get to name it once you save it.
+
+> Left of the Explorer you can see the so called _Activity Bar_ marked in red
+> below. By right clicking this you can change its contents or position on the
+> screen. 
```
godt

The1Penguin commented on src/beginner/editor/basics.md:
```diff
@@ -0,0 +1,88 @@
+something like open a folder, create a file or to save now we have to instead
+use the command palette or the menu bar.
+
+{{ .macos }} On MacOS the menu bar will look like this
+![MacOS menu bar](/assets/editor/menubar-macos.png)
+
+{{ .windows .linux }} On Windows and Linux the menu bar will look something
+like this
+![Windows menu bar](/assets/editor/menubar-lindows.png)
```
Uppskattar detta

The1Penguin commented on src/beginner/editor/extensions.md:
```diff
@@ -0,0 +1,84 @@
+website](https://vscodethemes.com/) allows you to see and compare different
+ones easily.
+
+<!-- it pains me not to mention catppuccin 😔-->
```
Do it ^^

The1Penguin commented on src/beginner/cli/basics.md:
```diff
@@ -0,0 +1,121 @@
+screen. You can also open one by clicking on `Terminal` in the menu bar at the
+top of your screen, and then pressing `New Terminal`.
+
+With a terminal open, what are we looking at? At the start of the line of text
+you will see your username followed by `@`, and then the name of your computer.
+After that you will see the current directory. Your terminal keeps track of
+something called its _working directory_, which is where on your file system it
+is located at the moment. Right now it is `~`, which is a symbol used to
+represent your _home directory_. Every user has a home directory where all of
+their files are stored. At the end of the line you will see a solid block,
+which is your cursor. Here we can write commands and run them by pressing
+!kbd[Enter].
```
Kanske lägga till en kommentar här om att det kan se annorlunda ut (typ nyinstall zsh) men detta är hur det vanligtvis ser ut för folk?

Multipacker replied:
Yes, vi har kollat oconfad bash och det är tillräckligt likt enligt oss, men ska kolla en gång till på några oconfade shells

Multipacker replied:
Har kollat oconfad bash och nyinstallation av fish och zsh i fedora och de följer alla den här beskrivningen, så jag tror det är relativt fine, lägger till en kommentar också

The1Penguin replied:
Aight, zsh var fucky för mig oconfad
<img width="314" height="56" alt="image" src="https://github.com/user-attachments/assets/9d186318-fec3-4638-9f1b-af6826ab5281" />

Multipacker replied:
Yeah vill minnas att jag har sett detta också men va osäker, weird. Är det hostnamet du får då?

The1Penguin replied:
Mjapp

Multipacker replied:
Hmm min default `/etc/zshrc` mappar explicit om ens prompt när den är satt till bara hostname

The1Penguin commented on src/beginner/editor/shortcuts.md:
```diff
@@ -1,21 +1,43 @@
+The entire line gets cut, meaning it is put in the clipboard and can be pasted
+elsewhere.
+
+![cheatsheet](/assets/editor/cli_editor_cheatsheet.png)
```
Kanske bara ha editor grejorna här?

The1Penguin replied:
Ni har kommandon separat i cli.commands

The1Penguin commented on src/beginner/cli/commands.md:
```diff
@@ -0,0 +1,61 @@
+
+If you want to know more about a command you can either use `command --help` to print a quick reference, or use `man <command>` to get more thorough documentation. The latter option can be a bit unintuitive to use at first, but you can use the arrow keys to move up and down, and !kbd[q] to exit out of the viewer.
+
+| Command                                 | Description                                                              |
+|-----------------------------------------|--------------------------------------------------------------------------|
+| `cat <files>`                           | Print the contents of the `<files>`                                      |
+| `cd <path>`                             | Changes your working directory to `<path>`                               |
+| `cp --recursive <source> <destination>` | Copies directories and their contents from `<source`> to `<destination>` |
+| `cp <source> <destination>`             | Copies files from `<source>` to `<destination>`                          |
+| `ls <path>`                             | Lists the files in `<path>`                                              |
+| `ls`                                    | Lists the files in the working directory                                 |
+| `mkdir <directories>`                   | Create the `<directories>`                                               |
+| `mv <source> <destination>`             | Moves files or directories from `<source>` to `<destination>`            |
+| `printenv`                              | Print all of your environment variables                                  |
+| `rm -r <directories>`                   | Removes the `<directories>` and all of their contents                    |
+| `rm <files>`                            | Removes the `<files>`                                                    |
+| `touch <files>`                         | Creates the `<files>`                                                    |
```
Se till att denna reflekterar cheetsheetet

Multipacker replied:
Fixat

The1Penguin commented on src/beginner/cli/powershell.md:
```diff
@@ -0,0 +1,56 @@
+covered earlier. Note that they only operate on single files, compared to Unix
+shells where they operate on multiple.
+
+| PowerShell Command                          | Unix like aliases                    | Description                                                              |
+|---------------------------------------------|--------------------------------------|--------------------------------------------------------------------------|
+| `Get-Content <file>`                        | `cat <file>`                         | Print the contents of the `<file>`                                       |
+| `Set-Location <path>`                       | `cd <path>`                          | Changes your working directory to `<path>`                               |
+| `Copy-Item <source> <destination>`          | `cp -Recurse <source> <destination>` | Copies directories and their contents from `<source`> to `<destination>` |
+| `Copy-Item -Recruse <source> <destination>` | `cp <source> <destination>`          | Copies files from `<source`> to `<destination>`                          |
+| `Get-ChildItem <path>`                      | `ls <path>`                          | Lists the files in `<path>`                                              |
+| `Get-ChildItem`                             | `ls`                                 | Lists the files in the working directory                                 |
+| `New-Item -ItemType directory <directory>`  | `mkdir <directory>`                  | Create the `<directory>`                                                 |
+| `Move-Item <source> <destination>`          | `mv <source> <destination>`          | Moves files or directories from `<source>` to `<destination>`            |
+| `Remove-Item <path>`                        | `rm <path>`                          | Removes the content at `<path>` and any potential child items            |
+| `New-Item <file>`                           |                                      | Creates the `<file>`                                                     |
+| `Get-Help <command>`                        |                                      | Prints the help page for `<command>`                                     |
+| `Get-Help <command> -Online`                |                                      | Opens the help page for `<command>` in a webbrowser                      |
```
```suggestion
| PowerShell Command                          | Unix like aliases                       | Description                                                              |
|---------------------------------------------|--------------------------------------------------------------------------------------------------------------------|
| `Get-Content <file>`                        | `cat <file>`                            | Print the contents of the `<file>`                                       |
| `Set-Location <path>`                       | `cd <path>`                             | Changes your working directory to `<path>`                               |
| `Copy-Item <source> <destination>`          | `cp <source> <destination>`             | Copies files from `<source`> to `<destination>`                          |
| `Copy-Item -Recruse <source> <destination>` | `cp --recursive <source> <destination>` | Copies directories and their contents from `<source`> to `<destination>` |
| `Get-ChildItem <path>`                      | `ls <path>`                             | Lists the files in `<path>`                                              |
| `Get-ChildItem`                             | `ls`                                    | Lists the files in the working directory                                 |
| `Move-Item <source> <destination>`          | `mv <source> <destination>`             | Moves files or directories from `<source>` to `<destination>`            |
| `Remove-Item <path>`                        | `rm <path>`                             | Removes the content at `<path>` and any potential child items            |
| `New-Item <file>`                           | `touch <file>`                          | Creates the `<file>`                                                     |
| `New-Item -ItemType directory <directory>`  | `mkdir <directory>`                     | Create the `<directory>`                                                 |
| `Get-Help <command>`                        | `man <command>                          | Prints the help page for `<command>`                                     |
| `Get-Help <command> -Online`                |                                         | Opens the help page for `<command>` in a browser                         |
```

Multipacker replied:
Vi har tolkat tabellhuvudet olika ser jag. Tanken va att PowerShell aliasar många av sina kommandon till unix versionen, så jag ville ha med hur det ser ut när man använder de aliasen. Det kanske blir förvirrande och att det ät bättre att skriva ut kommandona som vi visade dem tidigare i boken.

Multipacker replied:
Bra dock att ha dem i samma ordning som i den andra tabellen.

The1Penguin replied:
Men även med det så blir det fel med `-Recurse` delen. Jag tycker det är mer lärorikt att jämföra med de tidigare introducerade kommandona

Multipacker replied:
Ja det blir bättre att jämföra med de tidigare kommandona

The1Penguin commented on src/studies/eduroam.md:
```diff
@@ -0,0 +1,245 @@
+
+<details>
+<summary>Certificates</summary>
+<!-- mdbook turns of markdown completely when you use inline HTML, so this is
+the HTML used for code blocks, except it somehow automatically inserts the copy
+button. -->
```
Yippie

Multipacker replied:
mdbook smhi

The1Penguin reviewed (commented):
Looks good. Fixa de små kommentarerna imo (borde inte ta så lång tid), och sen mergea

Multipacker committed:
Remove Unix command for printing environment variables

Multipacker committed:
Fixed commmand listings

Multipacker committed:
Added a note about different prompts

Multipacker committed:
Updated getting started for Eduroam

Multipacker committed:
Formatted the introduction

Multipacker committed:
Eduroam consistency

SwedishSubmarine committed:
Added basic configuration for git

SwedishSubmarine committed:
Changed a weird sentence in eduroam

SwedishSubmarine committed:
Fixed flags a bit

Multipacker committed:
Added new cli commands pdf

SwedishSubmarine committed:
Formatting for commands

SwedishSubmarine committed:
Center horizon for description

SwedishSubmarine committed:
Reference to download commands.pdf

Multipacker committed:
Added new cheatsheets from typst

SwedishSubmarine committed:
Yay i love referencing MY theme

The1Penguin commented on README.md:
```diff
@@ -1,7 +1,20 @@
+Please see [the book](https://data101.dtek.se/contributing.html) for
+instructions on how to build and contribute to the project.
+
+## Compiling cheatsheets
+
+To compile the cheatsheets, use `./typst_compile.sh`. Make sure to commit the
+produced PDFs as this step is not included in the docker-file.
```
Varför inte göra det som ett step av bygget?
